docs/data/data-grid/filtering/ServerFilterGrid.tsx

```diff
@@ -1,83 +1,39 @@
 import * as React from 'react';
-import {
-  DataGrid,
-  GridRowModel,
-  GridFilterModel,
-  GridColumns,
-  getGridStringOperators,
-} from '@mui/x-data-grid';
+import { DataGrid, GridFilterModel } from '@mui/x-data-grid';
+import { serverConfiguration, UseDemoDataOptions } from '@mui/x-data-grid-generator';
 
-function loadServerRows(commodityFilterValue?: string): Promise<any> {
-  const serverRows = [
-    { id: '1', commodity: 'rice' },
-    { id: '2', commodity: 'soybeans' },
-    { id: '3', commodity: 'milk' },
-    { id: '4', commodity: 'wheat' },
-    { id: '5', commodity: 'oats' },
-  ];
+const DATASET_OPTION: UseDemoDataOptions = {
```

Comment:
To be consistent with `SERVER_OPTIONS`. Same for other demos.

```suggestion
const DATASET_OPTIONS: UseDemoDataOptions = {
```